core/src/services/routing.js

```diff
@@ -405,3 +409,27 @@ export const handleRouteClick = (node, windowElem = window) => {
   const route = buildRoute(node, `/${node.pathSegment}`);
   navigateTo(route, windowElem);
 };
+
+export const getModifiedHash = s => s.newURL.split('#/')[1];
+export const getModifiedPathname = () =>
+  window.history.state.path
+    .split('/')
+    .slice(1)
+    .join('/');
+
+export const getCurrentPath = () =>
+  getConfigValue('routing.useHashRouting')
+    ? window.location.hash.replace('#', '')
+    : trimLeadingSlash(window.location.pathname);
+
+export const addRouteChangeListener = callback => {
+  if (getConfigValue('routing.useHashRouting')) {
+    return window.addEventListener('hashchange', event => {
+      callback(getModifiedHash(event));
+    });
+  }
+
+  return window.addEventListener('popstate', () => {
```

Comment:
Don't we have to wrap this piece of code in an if statement, since it is specific to path routing? Something like:
```
if (!getConfigValue('routing.useHashRouting')) {
...
}
```

Reply:
No, if it is hashRouting, it will not get to this point due to `return window.addEventListener('hashchange'`.